src/macros.rs

```diff
@@ -230,6 +232,10 @@ fn rewrite_macro_inner(
         if let success @ Some(..) = format_lazy_static(context, shape, ts.clone()) {
             return success;
         }
+    } else if macro_name == "matches!" && context.config.version() == Version::Two {
+        if let success @ Some(..) = format_matches(context, shape, &macro_name, mac) {
+            return success;
+        }
```

Comment:
@calebcartwright per your comment https://github.com/rust-lang/rustfmt/pull/5554#issuecomment-1597396653, I just wanted to let you know that the changes are version gated.

I don't think there's a rush to get this in, and we can definitely hold off until `style_edition` lands to gate this around the `2024` style edition.